### ISSUE TITLE
Package Request: MGBA-MEDUSA, CITRA, maybe VULKAN/EPOXY?

### ISSUE THREAD

**Issue by Jakecheslekov** [voidlinux/void-packages#12988]
### System
Void Linux x64 MUSL
* xuname:  
  *output of ``xuname`` (part of xtools)*
Void 4.16.0rc7_1 x86_64-musl GenuineIntel uptodate rDF
* package:  
  *affected package(s) including the version*
MGBA-MEDUSA, CITRA & maybe also VULKAN.

MGBA's MEDUSA(Nintendo DS emulator):
https://codeload.github.com/mgba-emu/mgba/tar.gz/medusa-a2
https://mgba.io/downloads.html

CITRA (Nintendo 3DS emulator):
https://github.com/citra-emu/citra
https://github.com/citra-emu/citra/wiki/Building-For-Linux

I've tried to compile these packages from source but I've ran into two obvious issues: Vulkan & EPOXY. Not sure why but when running cmake for Medusa it doesn't seem to be able to find EPOXY even though libepoxy & libepoxy-devel are installed, and there are some errors about Vulkan during the make command too.

I'm going to post the pastebins in a few minutes.

**Comment by Jakecheslekov** [comment]
I've changed the title because that part of the issue actually had nothing to do with Void's packages, improving accuracy of the post.

**Comment by Jakecheslekov** [comment]
For the 
distfiles=
rule in template, how do I get it to pull a git repo instead of downloading a tar.gz ?

**Comment by Johnnynator** [comment]
take a look at the alacritty template. (or generally search templates with the _commit variable inside them)

**Comment by Jakecheslekov** [comment]
Um.... the alacritty template has a tar.gz, not a .git

**Comment by Johnnynator** [comment]
Yes that is intended. Github gives you a .tar.gz for every commit. 
E.g https://github.com/citra-emu/citra/archive/74465389f39431481465636314738a08bb3d13ec.tar.gz would give you a tar ball for citra

**Comment by Jakecheslekov** [comment]
I get a 404 when I try to go to github.com/citra-emu/citra/archive/ though, how am I supposed to know the commit number?

**Comment by Johnnynator** [comment]
also cmake and qmake are handled by the build_style variable. `build_style=cmake` is often enough (or `build_style=qmake`) for getting the program build.

You can find the commits number by clicking on the commits on [github](https://github.com/citra-emu/citra/commits/master)

**Comment by Johnnynator** [comment]
for the checksum I would recommend installing the xtools package. The is a tool called xgensum in it which gives filles in the checksum. So when you have the distfile filled in correctly you can run `xgensum -f -i srcpkgs/citra/template` and it should update it.

**Comment by Jakecheslekov** [comment]
I'd rather it be more straight-forward & just do it like 
cmake
make -j4
make install

instead of complicating it.... How do I add the variables to cmake or qmake now if I were to use "build_style" ?

**Comment by Johnnynator** [comment]
you add variables to cmake by adding `configure_args=` and to make by `make_build_args=` and to make install with `make_install_args=`.
Also have you read through the [Manual](https://github.com/voidlinux/void-packages/blob/master/Manual.md) since it explain a lot of such stuff.

**Comment by Jakecheslekov** [comment]
In alacritty, mgba, vba-m, etc. I don't see how I would add build flags to 'make' either like how most AUR packages are... Guess I'm just too stuck in my ways making basic PKGBUILDs...

**Comment by Johnnynator** [comment]
You can also try to make it without using build_style first to get it running. Changing the build_style afterwards wouldn't be hard.

**Comment by Jakecheslekov** [comment]
Do I have to use these _args= things? Can't I just do it the normal, less complicated way so it will actually show 
"cmake .."
"make -j4"
"make install" 
?

I really don't see the point in making these scripts harder to create by having to learn a whole new method...

**Comment by Johnnynator** [comment]
Using the build_style gives the advantage of being able to easily cross compile packages. Take a look at the cmake build style script in common/build-style/cmake.sh. You don't want every template to be that long and maintain it independently in every template.

**Comment by Jakecheslekov** [comment]
Like when all that has to be done is 

git clone https://github.com/mgba-emu/mgba.git -b medusa medusa
cd medusa
mkdir -p build && cd build
cmake ..
make -j4
make install

I don't see a point in complicating this process with all of these _args= things.... I'll read the wiki you linked but I don't know if I'll be able to make these templates any time soon so if anyone else really wants to do so then they can go ahead & do it.

**Comment by Johnnynator** [comment]
I quickly made an template for medusa, but don't expect it to be included into the void repo before the first release of medusa.
https://github.com/Johnnynator/void-packages/commit/5f17c865fb949101d1263c1ad783cfb78793d75e

**Comment by Johnnynator** [comment]
Also it is the same with citra which also doesn't have any tagged release.

**Comment by Jakecheslekov** [comment]
A pre-release wouldn't qualify it to get into the Void package manager? Pretty sure I've seen other pre-release software in the Void repo...
And how is it you can get that template to work when you've commented out the _args= ???
This is why these templates make no sense to me.

**Comment by Johnnynator** [comment]
I forgot to update the link to the template. 
https://github.com/Johnnynator/void-packages/commit/756e63df96e49fc1c64700f48120f64185b2a6e7

**Comment by Jakecheslekov** [comment]
I'm still way too confused on how these templates work... What part of the template runs make? What part of the template runs make install? I don't get it.. Don't understand why its not like PKGBUILD where I can just write the commands to compile it, I don't really think these templates abide by the "KISS" principle at all... And what does this 'vmove' command do??

**Comment by Jakecheslekov** [comment]
I tried making a template with just

git clone https://github.com/mgba-emu/mgba.git -b medusa medusa && cd medusa
mkdir -p build && cd build
cmake ..
make -j8 
sudo make install

but it ended up giving me:
=> ERROR: medusa-0.1_1: missing do_install() function!

what is do_install ?????

**Comment by Johnnynator** [comment]
xbps templates have a few stages, fetch, extract, configure, build, install and pkg, which are divided into 3 "sub parts" which are prefixed by pre_, do_ and post_. By specifying a build_style= you import a few predefined stages.
```
do_fetch(){} # fetches everything in distfiles. Hast to be overwritten when you want to use git clone
do_extract(){} # used for extracting files. normally just extracts everything in distfiles, has to be modified normally never
do_configure(){} # used for the configure phase
do_build(){} # normally used for the build phase
do_install(){} #always required
do_pkg() {} # never has to be messed with
```

```
do_fetch() {
git clone https://github.com/mgba-emu/mgba.git -b medusa $wrksrc
}
do_configure() {
mkdir -p build && cd build
cmake ..
}
do_build() {
cd build
make ${makejobs}
}
do_install() {
cd build
make install
}
```

**Comment by Johnnynator** [comment]
> I'm still way too confused on how these templates work... What part of the template runs make? What part of the template runs make install? 

If a build_style is specified it does use the do_{configure,build,install} functions of the build_style ( `common/build-style/` ). which does run make; make install with the mostly needed parameters.

**Comment by Jakecheslekov** [comment]
Well I tried to inquire on your forums to find out how these templates work but Duncaen is a little bitch who blocks people from being able to reply because he got too offended with the questions I was asking! 
I was asking if I could get away with just doing it the PKGBUILD way with the build commands & if not, why not, so I could get a better idea of how it works, but instead he calls me stubborn & starts hurling insults / getting mad at an analogy I made. Really pathetic, and then he stops me from being able to reply after he writes a big long post crying about how I ask questions.

**Comment by Jakecheslekov** [comment]
How dare I criticise the Void Linux templates by saying PKGBUILDs are better because of simplicity, that must be a bannable offense LMFAO! You people are fucking pathetic, I'm going back to Gentoo, fucking coward crybaby assholes can't handle the slightest bit of criticism.

**Comment by pullmoll** [comment]
Read https://github.com/voidlinux/void-packages/blob/master/CONTRIBUTING.md and also https://github.com/voidlinux/void-packages/blob/master/Manual.md to understand how Void Linux templates work.

If you don't want to or can't, your templates probably won't be accepted.
Using foul language certainly doesn't help to get your questions answered.
This part isn't described in the manuals but called common sense.

**Comment by Jakecheslekov** [comment]
I don't really give a flying fucking shit, I'm not contributing to a bunch of crybaby coward assholes who can't handle the slightest bit of criticism & ban anyone off their forums who tries to ask questions & learn, go fuck yourselves.
